packages/salesforcedx-vscode-core/src/commands/forceAuthDevHub.ts

```diff
@@ -37,6 +52,55 @@ export class ForceAuthDevHubExecutor extends SfdxCommandletExecutor<{}> {
     command.withArg('--setdefaultdevhubusername');
     return command.build();
   }
+
+  public async execute(response: ContinueResponse<any>): Promise<void> {
+    const cancellationTokenSource = new vscode.CancellationTokenSource();
+    const cancellationToken = cancellationTokenSource.token;
+
+    const execution = new CliCommandExecutor(this.build(response.data), {
+      cwd: getRootWorkspacePath()
+    }).execute(cancellationToken);
+
+    execution.processExitSubject.subscribe(async () => {
+      const globalDevHubName = await this.getDevNubHame(ConfigSource.Global);
+      if (isNullOrUndefined(globalDevHubName)) {
+        const localDevHubName = await this.getDevNubHame(ConfigSource.Local);
+        if (isNullOrUndefined(localDevHubName) === false) {
+          await this.setGlobal(String(localDevHubName));
+        }
+      }
+    });
+
+    this.attachExecution(execution, cancellationTokenSource, cancellationToken);
+  }
+
+  private async setGlobal(newUsername: string) {
```

Comment:
Can we make this method name more descriptive? what global value are we setting?